torchquad/integration/gaussian.py

```diff
@@ -0,0 +1,209 @@
+        diff = self._integration_domain.T[1] - self._integration_domain.T[0]
+        return 0.5*diff
+        
+        
+class GaussLaguerre(Gaussian):
+    """Gauss Laguerre quadrature rule in torch, for integrals of the form :math:`\int_0^{\infty} e^{-x} f(x) dx`. It will correctly integrate polynomials of degree :math:`2n - 1` or less
+    over the interval :math:`[0, \infty]` with weight function
+    :math:`f(x) = x^{\alpha} e^{-x}`. See https://en.wikipedia.org/wiki/Gauss%E2%80%93Laguerre_quadrature.
+    
+    Examples
+    --------
+    >>> gl=torchquad.GaussLaguerre()
+    >>> integral=gl.integrate(lambda x,a: np.sin(a*x),dim=1,N=20,args=(1,)) #integral from 0 to inf of np.exp(-x)*np.sin(x)
+    |TQ-INFO| Computed integral was 0.49999999999998246. #analytic result = 0.5"""
+
+    def __init__(self):
+        super().__init__()
+        self.name="Gauss-Laguerre"
+        self.root_fn=scipy.special.roots_laguerre
+        self.default_integration_domain=[[0,numpy.inf]]
+        self.wrapper_func=None
+        
+    def _get_constant_multiplier(self):
+        return 1.0
+        
+        
+class GaussHermite(Gaussian):
+    """Gauss Hermite quadrature rule in torch, for integrals of the form :math:`\int_{-\infty}^{+\infty} e^{-x^{2}} f(x) dx`. It will correctly integrate
+    polynomials of degree :math:`2n - 1` or less over the interval
+    :math:`[-\infty, \infty]` with weight function :math:`f(x) = e^{-x^2}`. See https://en.wikipedia.org/wiki/Gauss%E2%80%93Hermite_quadrature
+    
+    Examples
+    --------
+    >>> gh=torchquad.GaussHermite()
+    >>> integral=gh.integrate(lambda x: 1-x,dim=1,N=200) #integral from -inf to inf of np.exp(-(x**2))*(1-x)
+    |TQ-INFO| Computed integral was 1.7724538509055168. #analytic result = sqrt(pi)
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.name="Gauss-Hermite"
+        self.root_fn=scipy.special.roots_hermite
+        self.default_integration_domain=[[-1*numpy.inf,numpy.inf]]
+        self.wrapper_func=None
+    
+    def _get_constant_multiplier(self):
+        return 1.0
+                
+                
+#class GaussLobatto(Gaussian):
+#    """Gauss Lobatto quadrature rule in torch. See """
+#
+#    def __init__(self):
+#        super().__init__()
+#        self.name="Gauss-Lobatto"
+#        self.root_fn=scipy.special.roots_jacobi #need additional function to operate on xi,wi to get the correct weights and points from the Jacobi polynomial roots...
+#
+#    def wrapper_func(self,xi,wi): #sould override default None
+
+
+#class GaussRadau(Gaussian):
+#    """Gauss Radau quadrature rule in torch. See """
+#
+#    def __init__(self):
+#        super().__init__()
+#        self.name="Gauss-Lobatto"
+#        self.root_fn=scipy.special.roots_jacobi #need additional function to operate on xi,wi to get the correct weights and points from the Jacobi polynomial roots...
```

Comment:
Maybe we can describe the missing thing with this code snippet in the associated issue in case we want to ad dthis later? Just to not have dead code in there